packages/components/nodes/chains/ApiChain/OpenAPIChain.ts

```diff
@@ -0,0 +1,84 @@
+import { ICommonObject, INode, INodeData, INodeParams } from '../../../src/Interface'
+import { APIChain, createOpenAPIChain } from 'langchain/chains'
+import { CustomChainHandler, getBaseClasses } from '../../../src/utils'
+import { ChatOpenAI } from 'langchain/chat_models/openai'
+
+class OpenApiChain_Chains implements INode {
+    label: string
+    name: string
+    type: string
+    icon: string
+    category: string
+    baseClasses: string[]
+    description: string
+    inputs: INodeParams[]
+
+    constructor() {
+        this.label = 'OpenAPI Chain'
+        this.name = 'openApiChain'
+        this.type = 'openApiChain'
+        this.icon = 'openapi.png'
+        this.category = 'Chains'
+        this.description = 'Chain to run queries against OpenAPI'
+        this.baseClasses = [this.type, ...getBaseClasses(APIChain)]
+        this.inputs = [
+            {
+                label: 'ChatOpenAI Model',
+                name: 'model',
+                type: 'ChatOpenAI'
+            },
+            {
+                label: 'YAML File',
+                name: 'yamlFile',
+                type: 'file',
+                fileType: '.yaml'
+            },
+            {
+                label: 'Headers',
+                name: 'headers',
+                type: 'json',
+                additionalParams: true,
+                optional: true
+            }
+        ]
+    }
+
+    async init(nodeData: INodeData): Promise<any> {
+        const model = nodeData.inputs?.model as ChatOpenAI
+        const headers = nodeData.inputs?.headers as Record<string, string>
```

Comment:
For `headers`, you have to parse it because it is passed as string

`const headers = nodeData.inputs?.headers as string`

then pass to `createOpenAPIChain`:
`headers: typeof headers === 'object' ? headers : headers ? JSON.parse(headers) : {}`

Reply:
My bad I used auto suggested copilot code should have taken a look at it will fix it.Good catch.